packages/n4s/src/rules/isArrayOf.js

```diff
@@ -0,0 +1,4 @@
+export function isArrayOf(value, ...rules) {
+    if (value.some(element => !rules.some(fn => fn(element)))) {return false};
+    return true;
+  };
```

Comment:
There are a couple of things I would consider changing in this function.

1. value might be `undefined`, and in some cases that would be ok. I would add an initial isArray safeguard at the beginning, and if value is not an array, return false immediately. At the moment, this function would throw.

2. I think you could skip the negation (!) on `!rules.some` if you switch `value.some` with `value.every`.

3. The inner function `fn => fn(element)` wouldn't work anymore due to a [recent change](https://github.com/ealush/vest/commit/40591805dcf5b00bfc005895eb0303491c66958a#diff-4e3cd2887d1ed9da69b47db120ccee942a92e9837efc21d70fc76409374b61afR80) I made to the lazy rules. They now support chaining as well, and to run them you need to call `.test`. You could use the `runLazyRules(fn, element)`

```suggestion
import runLazyRules from 'runLazyRules';

export function isArrayOf(value, ...rules) {
  return !value.every(element => rules.some(ruleChain => runLazyRules(ruleChain, element)))
};
```